File: 1502. Can Make Arithmetic Progression From Sequence/solution.py
class Solution(object):
  def canMakeArithmeticProgression(self, arr):
    """
    :type arr: List[int]
    :rtype: bool
    """
    arr.sort()
    difference = arr[1] - arr[0]
    
    for index, value in enumerate(arr[:-1]):
      nextValue = arr[index + 1]
      if nextValue - value != difference:
        return False
    
    return True

File: 1502. Can Make Arithmetic Progression From Sequence/solution.cpp
class Solution {
public:
  bool canMakeArithmeticProgression(vector<int>& arr) {
    sort(arr.begin(), arr.end());
    int difference = arr[1] - arr[0];

    for (int index = 0; index < arr.size() - 1; index++){
      int value = arr[index];
      int nextValue = arr[index + 1];
      if (nextValue - value != difference){
        return false;
      }
    }

    return true;
  }
};